### PR TITLE
Add `Timestamptz` support for the `time` crate v0.3, to Diesel's 2.0 branch

### PR DESCRIPTION
**Pull request by coriolinus**
As requested in https://github.com/diesel-rs/diesel/discussions/3241

### CODE REVIEW

**Comment by weiznich** [comment]
I was just about to look into the test failures. Glad to see that this is already fixed. Seems to be a subtle difference in how time and chrono formates timestamps with millisecond resolution.

**Comment by coriolinus** [comment]
Yes, chrono gives you a format flag `%.f` which emits subseconds only if non-zero, whereas with the time crate you have to emulate that manually. It's not a huge deal once I figured out what the issue actually was; most of the debugging involved was just in figuring out what was actually going wrong.

File: diesel/Cargo.toml [hunk 1]
```diff
@@ -33,6 +33,9 @@ bigdecimal = { version = ">=0.0.13, < 0.4.0", optional = true }
 bitflags = { version = "1.2.0", optional = true }
 r2d2 = { version = ">= 0.8.2, < 0.9.0", optional = true }
 itoa = { version = "1.0.0", optional = true }
+# newer versions of `time` have a MSRV of 1.57.0, which is newer than
+# diesel's own MSRV of 1.56.0
+time = { version = "=0.3.9", optional = true, features = ["macros", "formatting", "parsing"] }
```
Comment (weiznich) [inline diesel/Cargo.toml:38]: It is fine to also allow newer time versions here, as we define our MSRV not by using the latest variant of all dependencies, but the oldest one. Or better: There must be a dependency version that builds fine with the given MSRV, we do not require that to be true for all dependency versions.

Comment (weiznich) [reply diesel/Cargo.toml:38]: Additionally we want to include `time` in the `extras` feature in line 52

**Review by weiznich — COMMENTED** [review]
This looks good for me. I've noted a few minor things regarding to how the optional dependency is setup and tested. If those are fixed, this is fine for merging.

**Pull request merged**